src/e2e-test/java/io/cdap/plugin/odp/stepsdesign/RuntimeODP.java

```diff
@@ -0,0 +1,422 @@
+    gcpClient.dropBqQuery(CDAPUtils.getPluginProp(table));
+    BeforeActions.scenario.write("Table Deleted Successfully");
+  }
+
+  @Then("link source and target")
+  public void linkSourceAndTarget() {
+    SeleniumHelper.dragAndDrop(ODPLocators.fromODP, CdfStudioLocators.toBigQiery);
+  }
+
+  @Then("Delete all existing records in datasource")
+  public void deleteAllExistingRecordsInDatasource() throws JCoException, IOException {
+    sapProps = SAPProperties.getDefault(connection);
+    errorCapture = new ErrorCapture(exceptionUtils);
+    sapAdapterImpl = new SAPAdapterImpl(errorCapture, connection);
+    Map opProps = new HashMap<>();
+    opProps.put("RFC", "ZTEST_DATA_DEL_ACDOCA");
+    opProps.put("autoCommit", "true");
+    try {
+      ObjectMapper mapper = new ObjectMapper();
+      ObjectNode objectNode = mapper.createObjectNode();
+      objectNode.put("I_ALL", "X");
+      JsonNode response = sapAdapterImpl.executeRFC(objectNode.toString(), opProps,
+                                                    StringUtils.EMPTY, StringUtils.EMPTY);
+    } catch (Exception e) {
+      throw SystemException.throwException(e.getMessage(), e);
+    }
+  }
+
+  public static void main(String args[]) throws IOException, JCoException, InterruptedException {
+    RuntimeODP runtimeODP = new RuntimeODP();
+    runtimeODP.deleteTheExistingOdpTableInBigquery("tab_src01");
+  }
+
+  @Then("{string} the {string} records with {string} in the ODP datasource from JCO")
+  public void createTheRecordsInTheODPDatasourceFromJCO(String process, String recordcount, String rfcName)
+    throws IOException, JCoException {
+    action = process;
+    if (action.equalsIgnoreCase("create")) {
+      action = "I_NUM_C";
+    } else if (action.equalsIgnoreCase("delete")) {
+      action = "I_NUM_D";
+    } else if (action.equalsIgnoreCase("update")) {
+      action = "I_NUM_U";
+    }
+    dsRecordsCount = Integer.parseInt(recordcount);
+    sapProps = SAPProperties.getDefault(connection);
+    errorCapture = new ErrorCapture(exceptionUtils);
+    sapAdapterImpl = new SAPAdapterImpl(errorCapture, connection);
+    Map opProps = new HashMap<>();
+    opProps.put("RFC", CDAPUtils.getPluginProp(rfcName));
+    opProps.put("autoCommit", "true");
+    try {
+
+      ObjectMapper mapper = new ObjectMapper();
+      ObjectNode objectNode = mapper.createObjectNode();
+      objectNode.put(action, recordcount);
+      JsonNode response = sapAdapterImpl.executeRFC(objectNode.toString(), opProps, "", "");
+      noOfRecords = Integer.parseInt(response.get("EX_COUNT").asText());
+      Iterator<JsonNode> iteratedData = response.get("EX_DATA").iterator();
+      while (iteratedData.hasNext()) {
+        JsonNode object = iteratedData.next();
+        int zeroIndex = 0;
+        Iterator<String> fieldName = object.fieldNames();
+        while (fieldName.hasNext() && zeroIndex == 0) {
+          numList.add(object.get(fieldName.next()).asText());
+          zeroIndex++;
```

Comment:
if (fieldName.hasNext()) {
  numList.add(object.get(fieldName.next()).asText());
}

Reply:
since we have multiple elements inside the iterator, thats why while loop

Reply:
done